lib/storage/metadata/mongoclient/MongoClientInterface.js

```diff
@@ -919,19 +920,48 @@ class MongoClientInterface {
                                       params, log, cb);
     }
 
-    internalListObject(bucketName, params, log, cb) {
-        const extName = params.listingType;
-        const extension = new listAlgos[extName](params, log);
-        const requestParams = extension.genMDParams();
+    internalListObject(bucketName, params, extension, log, cb) {
         const c = this.getCollection(bucketName);
         let cbDone = false;
-        const stream = new MongoReadStream(c, requestParams,
+        let streakLength = 0;
+        const stream = new MongoReadStream(c, params,
                                            params.mongifiedSearch);
         stream
             .on('data', e => {
-                if (extension.filter(e) < 0) {
+                // filtering result:
+                // - filteringResult > 0: entry accepted and included in the
+                //                        listing result
+                // - filteringResult = 0: entry accepted but not included in
+                //                        the listing result (skipping)
+                // - filteringResult < 0: entry is rejected, listing finishes
+                const filteringResult = extension.filter(e);
+                // skipping: provides insight into why the extension is
+                // skipping the entry, for example: entry starts with a
+                // commonPrefix or entry is a specific version; in such
+                // case, repd needs to skip this range and go on with the next
```

Comment:
I certainly miss something but from the skipping implementation I see in lib/algos/list/, I don't understand how skipping can give this kind of information. It can return SKIP_NONE, '', undefined or nextMarker.